src/enrich/sled.rs

```diff
@@ -134,15 +134,11 @@ fn fetch_referenced_utxo<'a>(
 
 impl Worker {
     #[inline]
-    fn batch_insert_new_utxos(
-        &self,
-        db: &sled::Db,
-        txs: &[MultiEraTx],
-    ) -> Result<(), crate::Error> {
+    fn insert_produced_utxos(&self, db: &sled::Db, txs: &[MultiEraTx]) -> Result<(), crate::Error> {
         let mut insert_batch = sled::Batch::default();
 
         for tx in txs.iter() {
-            for (idx, output) in tx.outputs().iter().enumerate() {
+            for (idx, output) in tx.produces().iter().enumerate() {
                 let key: IVec = format!("{}#{}", tx.hash(), idx).as_bytes().into();
```

Comment:
This is a bug: if the Tx is invalid then a UTxO using the collateral return TxOut is created but the TxIn for the UTxO is defined as having the index `|txouts|`, that is the collateral return output is given the next available index after the number of txouts contained in the transaction, but the current implementation gives it the index `0`. This affects the UTxO reducer as well.